src/toil/resource.py

```diff
@@ -511,7 +529,10 @@ def _runningOnWorker(self):
             # we can reasonably assume that we are not running
             # on a worker node.
             try:
-                mainModuleFile = os.path.basename(mainModule.__file__)
+                if mainModule.__file__ is not None:
```

Comment:
I think we are trying to detect if there is no `__file__` set, and in that case return `False`. With this check we no longer get an exception so I think the function won't work anymore.